zdm/energetics.py

```diff
@@ -19,6 +21,35 @@ def init_igamma_splines(gammas, reinit=False):
             # iGamma
             igamma_splines[gamma] = interpolate.splrep(avals, numer)
 
+def init_igamma_linear(gammas, reinit=False, log=False):
+
+    for gamma in gammas:
+        
+        if log:
+            if gamma not in igamma_linear_log10.keys() or reinit:
+                print(f"Initializing igamma_linear for gamma={gamma} with log10")
+                 # values
+                avals = 10**np.linspace(-6, 6., 1000)
+
+                numer = np.array([float(mpmath.gammainc(
+                    gamma, a=iEE)) for iEE in avals])
+
+                # convert avals to log10 space (init x values)
+                log_avals = np.log10(avals)
+
+                igamma_linear_log10[gamma] = interpolate.interp1d(log_avals, numer)
+
+        else:
```

Comment:
If the only difference in the two paths of the `if/else` is line `38` then can we reduce it to just that line?  Or maybe it needs to be that line and the one below it that fills the `dict`.

Reply:
Agree, looks like the if/else should be only around line 38

Reply:
fixed